### PR TITLE
Add support for Ruby 2.3.0 and &.

### PR DESCRIPTION
**Pull request by presidentbeef**
This will require minor changes to the pt test cases as well, but otherwise this seems to work.

### CODE REVIEW

**Comment by presidentbeef** [comment]
I'm happy to update ruby2ruby as well, when there is agreement on the new node type (I called it `:safe_call` in this diff but I'm not particular).

File: lib/ruby_lexer.rex [hunk 1]
```diff
@@ -133,6 +133,7 @@ was_label?        /\'#{SSTRING}\':?/o   process_label_or_string
 |               /\&\&\=/                { result(:expr_beg, :tOP_ASGN, "&&") }
 |               /\&\&/                  { result(:expr_beg, :tANDOP,   "&&") }
 |               /\&\=/                  { result(:expr_beg, :tOP_ASGN, "&" ) }
+|               /\&\./                  { result(:expr_dot, :tLONELY,  "&.") }
```
Comment (zenspider) [inline lib/ruby_lexer.rex]: ugh

**Comment by zenspider** [comment]
Done. Thanks!

**Comment by presidentbeef** [comment]
Shoot. This won't parse `x&.y = z`.

**Pull request closed without merging**